src/MatrixTypes_PETSc.f90

```diff
@@ -565,6 +568,20 @@ SUBROUTINE matmult_PETSc(A,B,C,alpha,beta,transA,transB)
       DEALLOCATE(tmpB)
       DEALLOCATE(tmpC)
     ENDSUBROUTINE matmult_PETSc
+!
+!-------------------------------------------------------------------------------
+!> @brief set all the entries of the matrix to be zero
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE zeroentries_PETScMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='zeroentries_PETScMatrixType'
+      CLASS(PETScMatrixType),INTENT(INOUT) :: matrix
+      PetscErrorCode  :: ierr
+      IF(matrix%isInit) THEN
+        CALL MatZeroEntries(matrix%a,ierr)
```

Comment:
The call to this routine should be added to the `testTPLPETSc` unit test to insure the version of petsc that we are using supports this interface.